c/src/lib/libbsp/arm/shared/stm32fxxxx/spi/hal-spi.c

```diff
@@ -0,0 +1,507 @@
+               break;
+   case SPI_SIX:   return (IRQn_Type)GET_SPI_VEC(SPI_SIX++);
+               break;
+   }
+}
+
+
+/**
+ * SPI IOCTL
+ */
+static int stm32_spi_ioctl(
+  spi_bus *base,
+  uint32_t ioctl_cmd,
+  unsigned long arg
+)
+{
+   rtems_status_code sc = RTEMS_SUCCESSFUL;
+   stm32_spi_bus *bus = (stm32_spi_bus *) base;
+
+   _Assert(base != NULL);
+
+   switch(ioctl_cmd)
+   {
+      /* Two line half-duplex mode */
+      case SPI_IOCTL_DIR_2LINE:    __HAL_SPI_ENABLE(bus->handle);
+                           bus->handle->Instance->CR1 &= ~(SPI_CR1_BIDIMODE);
+                           __HAL_SPI_DISABLE(bus->handle);
+                           return 0;
+                           break;
+
+      /* bi-direction line full-duplex mode */
+      case SPI_IOCTL_DIR_1LINE:   __HAL_SPI_ENABLE(bus->handle);
+                           bus->handle->Instance->CR1 |= SPI_CR1_BIDIMODE;
+                           __HAL_SPI_DISABLE(bus->handle);
+                           return 0;
+                           break;
+
+      /* Configure SPI instance to Master mode */
+      case SPI_IOCTL_MODE_MASTER: __HAL_SPI_DISABLE(bus->handle);
+                           bus->handle->Instance->CR1 |= (SPI_CR1_MSTR | SPI_CR1_SSI);
+                           __HAL_SPI_ENABLE(bus->handle);\
+                           return 0;
+                           break;
+
+      /* Configure SPI instance to Slave mode */
+      case SPI_IOCTL_MODE_SLAVE:   __HAL_SPI_DISABLE(bus->handle);
+                           bus->handle->Instance->CR1 &= ~(SPI_CR1_MSTR | SPI_CR1_SSI);
+                           __HAL_SPI_ENABLE(bus->handle);
+                           return 0;
+                           break;
+
+      /* Enable CRC Calculation */
+      case SPI_IOCTL_ENABLE_CRC:   __HAL_SPI_DISABLE(bus->handle);
+                           bus->handle->Instance->CR1 |= SPI_CR1_CRCEN;
+                           __HAL_SPI_ENABLE(bus->handle);
+                           return 0;
+                           break;
+      /* Disable CRC Calculation */
+      case SPI_IOCTL_DISABLE_CRC:   __HAL_SPI_DISABLE(bus->handle);
+                           bus->handle->Instance->CR1 &= ~(SPI_CR1_CRCEN);
+                           __HAL_SPI_DISABLE(bus->handle);
+                           return 0;
+                           break;
+
+      default:               return -1;
+   }
+}
+
+/**
+ * SPI Interrupt Event ISR
+ */
+static void stm32_spi_event_irq(void* arg)
+{
+   rtems_status_code sc;
+   stm32_spi_bus *bus;
+
+   HAL_SPI_IRQHandler((SPI_HandleTypeDef *) arg);
+
+   sc = rtems_event_transient_send(bus->task_id);
+   _Assert(sc == RTEMS_SUCCESSFUL);
+}
+
+/**
+ * SPI Initiate Transfer
+ */
+static int stm32_spi_transfer(
+  spi_bus *base,
+  spi_msg *msgs,
+  uint32_t msg_count
+)
+{
+   rtems_status_code sc = RTEMS_SUCCESSFUL;
+   stm32_spi_bus *bus = (stm32_spi_bus *) base;
+   rtems_interval tickstart;
+
+   _Assert(base != NULL);
```

Comment:
I *think* this is a good place for an Assert. Assertions should be used for indicating a critical error that should never happen. In this case, it appears that base should never be null unless there is an internal error in the code for this module, or there is a CPU/memory fault.